### PR TITLE
pom improvements

### LINKED ISSUE #32: Fix problems building on JDK 11
Currently, the extension is built using JDK 11 and JDK 8 on GH.
The dokka plugin is not working JDK 11 (as far as I know), so the default for building should be JDK8, if we build dokka pages, or even better the creation of sources and javadoc should be moved to a separate profile used by release process only (which can then work on JDK 8).

### PR DESCRIPTION
**Pull request by zambrovski**
Following improvements:

- mvn Wrapper provides an easy way to be independent from installed maven version
- in parent pom, the "plugin management" defines all required plugins (including versions and configurations)
- the behavior of the "build" should not be inherited from the parent. The "pom" module doesn't need compiler plugin runs (this produces errors and warnings during build) - these are defined in "jar" modules only
- creation of Javadoc and Sources is moved to a profile inside the Kotlin module. It is automatically triggered during release but can be invoked as described in the readme too. 
- switched to latest version of dokka (1.4.0) which generates javadoc in JDK versions [8,11,...]
- added .editorconfig file, to ease the formatting of the code
- this will fix #32

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -35,11 +36,11 @@
     <kotlin.version>1.3.61</kotlin.version>
     <kotlin-logging.version>1.7.7</kotlin-logging.version>
     <jackson-kotlin.version>2.10.0</jackson-kotlin.version>
-    <mockk.version>1.9.3</mockk.version>
+    <mockk.version>1.10.0</mockk.version>
     <junit.jupiter.version>5.5.2</junit.jupiter.version>
     <slf4j.version>1.7.30</slf4j.version>
     <log4j.version>2.13.3</log4j.version>
-    <dokka.version>0.10.1</dokka.version>
+    <dokka.version>1.4.0</dokka.version>
```
Comment (sandjelkovic) [inline pom.xml:43]: Won't this also bump Kotlin version to 1.4?

Comment (smcvb) [reply pom.xml:43]: Oh, I just did push the extension to 1.4, as a follow up of dependabot, which I figured to be the right thing to do. If this was wrong, please correct me on it.

Comment (sandjelkovic) [reply pom.xml:43]: Kotlin 1.3 -> 1.4 is not a significant change code wise, so everything should be fine. That does mean that the extension will be dependent on Kotlin 1.4 from now on.

**Comment by zambrovski** [comment]
I believe we don't need Kotlin 1.4 yet, but just require Dokka 1.4 (not 1.4.10) since it solves the JDK 11 problem.

Independent of the compilation status - please let merge this PR and then fix the code if required, since this one is pure infrastructure and not related to programming. To get the build green we need correct infrastructure first and then fix all errors.

**Review by sandjelkovic — APPROVED** [review]
Looks good to me 👍

File: pom.xml [hunk 2]
```diff
@@ -254,211 +253,219 @@
...
+              <goals>
+                <goal>dokka</goal>
+                <goal>javadocJar</goal>
+              </goals>
+            </execution>
+          </executions>
+        </plugin>
+        <plugin>
+          <artifactId>maven-jar-plugin</artifactId>
+          <version>3.0.2</version>
```
Comment (lfgcampos) [inline pom.xml:364]: A downgrade here but seems like a merge mistake.

**Review by lfgcampos — APPROVED** [review]
LGTM

File: pom.xml (hunk 2 above)
Comment (zambrovski) [reply pom.xml]: fixed

Comment (zambrovski) [reply pom.xml]: dependabot will report all mistakes in downgrades, if any :-)

**Review by smcvb — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**